src/indexPage/endpoint.js

```diff
@@ -0,0 +1,65 @@
+import { get as getStorage, set as setStorage } from 'binary-common-utils/lib/storageManager';
+import { LiveApi } from 'binary-live-api';
+import { getWebSocketURL, getDefaultEndpoint, getAppIdFallback } from '../common/appId';
+
+export default function endpoint() {
+    if (!document.location.href.match(/endpoint\.html/)) return false;
+    $(document).ready(() => {
+        $('#error').hide();
+        $('#connected').hide();
+        let api;
+        const generateConnectedMessage = () => `<b>Connected to the Endpoint ${getStorage('config.server_url')}!</b>`;
+        const generateErrorMessage = () => `Unable to connect to ${getStorage('config.server_url')}. Switching connection to default endpoint.`;
+        const initConnection = () => {
+            if (api && api.disconnect) {
+                api.disconnect();
+            }
+            api = new LiveApi({
```

Comment:
It won't protect us from the cases where the endpoint is non-responsive.

I'd prefer something like this:

```
const api = getLiveApiInstance();
await api.ping();
setStorage('config.server_url', serverUrl);
...
```